### PR TITLE
Remove python-logstash dependency

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
It is more work to maintain this as a dependency than it is to just edit the code.

I do not want to upgrade this library (except for manually going through their diff), particularly as we go to python3, ping @ryanpetrello 

Will leave WIP in title here until I can verify functionality in a more complete way, which shouldn't take too long.

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME
 - API

##### AWX VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: awx/main/utils/formatters.py
Comment (AlanCoding) [inline awx/main/utils/formatters.py:23]: Previously, we inherited from `LogstashFormatterVersion1`, which inherited from `LogstashFormatterBase`.

However, `LogstashFormatterVersion1` only defined 1 method, `format`. We overwrote this method entirely.

File: awx/main/utils/formatters.py
Comment (ryanpetrello) [inline awx/main/utils/formatters.py:25]: @wenottingham I assume MIT means this is fine?

https://github.com/ansible/awx/blob/devel/docs/licenses/python-logstash.txt

**Review by ryanpetrello — APPROVED** [review]

File: awx/main/utils/formatters.py
Comment (wenottingham) [reply awx/main/utils/formatters.py:25]: ... I'd link directly to the license/project upstream. Having a dangling license text for something we don't include is going to run afoul of automation sooner or later.

**Comment by ryanpetrello** [comment]
Holding off on merging this for now.

**Comment by AlanCoding** [comment]
Is the intent to revive this after the python3 branch merges? Or did you have some other variant of that plan in mind?

File: awx/main/utils/formatters.py
Comment (ryanpetrello) [reply awx/main/utils/formatters.py:25]: ping @wenottingham, are there are any special doc/licensing considerations to make here if we're bundling this code from this external library?

**Review by ryanpetrello — APPROVED** [review]
If this is literally the only reason we depend on this library, then I'm fine with this.

Does keeping this library up to date cost us anything?  Are there any dependencies that we lose other than the library itself?  Just want to make sure I understand the motivation for this change before pushing the button.  Does 
https://github.com/ansible/awx/pull/2736#discussion_r234612028 capture why we want to do this?

File: awx/main/utils/formatters.py
Comment (wenottingham) [reply awx/main/utils/formatters.py:25]: No, we should just note where it comes from.

File: awx/main/utils/formatters.py
Comment (AlanCoding) [inline awx/main/utils/formatters.py:69]: we may want `stack_info` added to list

https://github.com/vklochan/python-logstash/issues/85

**Comment by AlanCoding** [comment]
> If this is literally the only reason we depend on this library, then I'm fine with this.

Yes, I feel very confident saying there is nothing else we were getting from that library.

> Does keeping this library up to date cost us anything? Are there any dependencies that we lose other than the library itself?

There were no requirements for the library either in a requirements file or `setup.py`.

That said... It did import `pika`, which was kind of like a land mine just hanging around our code that we happen to not hit.

```
(awx) bash-4.2$ python3
Python 3.6.6 (default, Aug 13 2018, 18:24:23) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-28)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from logstash.handler_amqp import AMQPLogstashHandler
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/venv/awx/lib64/python3.6/site-packages/logstash/handler_amqp.py", line 10, in <module>
    import pika
ModuleNotFoundError: No module named 'pika'
```

> Does #2736 (comment) capture why we want to do this?

That's one piece. There are a few things we were getting from it. The primary thing, IMO, was just kicking off JSON formatter class for us. Some of its blacklisted fields seem to relate specifically to logstash, and not the python logging library, so there's that. There were many other things that it offered that never mattered to us. I hope I've reduced it to the barebones. Some of the things it does might still not be done in the best way for us. There are other things that it did, but we already copy+pasted over to our own code.

I also want to do other things, like introduce a YAML formatter class. This has a advantage over JSON in that you can parse the entire file at any given point with standard pyyaml. I would actually like to build out some utilities to shell into my docker container, grab logs from a file, and organize them via a context manager while I do stuff via an AWX client library.

**Comment by ryanpetrello** [comment]
👍 if you'll clean up the flake8 failure, this looks mergeable to me.

File: awx/main/utils/formatters.py
Comment (AlanCoding) [reply awx/main/utils/formatters.py:25]: Good question, I'd also like @wenottingham to have a quick glance. Otherwise I'm ready to slap the "mergeit" on this.

Comment (wenottingham) [reply awx/main/utils/formatters.py:25]: by wfm: the way it's denoted there is fine. I haven't tested the code.

**Pull request merged**
